### PR TITLE
Project documentation for Ruxandra Popa

### PR DESCRIPTION
**Pull request by ruxi22**
### Pull Request Overview

This pull request adds significant improvements to the RustPlayer project:

- Lays out the initial project architecture and hardware component integration strategy for the RustPlayer, detailing the use of the Raspberry Pi Pico W, OLED display, buttons, and audio output components.
- Adds preliminary documentation, including schematics and proposed software libraries, essential for the development of the MP3 player.

### Testing Strategy

While the current phase of the RustPlayer project involves foundational documentation and preliminary planning, a comprehensive testing strategy is crucial for ensuring the reliability and functionality of the finished product. Here’s the planned approach for testing as the project progresses into the development and implementation stages:

1. Unit Testing:
 - Purpose: To test individual components and functions within the software to ensure they perform as expected.
 - Method: Write and execute unit tests for each software module using a framework suitable for embedded systems in Rust, such as cargo test. Focus on testing the interaction with the file system, user input processing, and audio output handling.
 
2. Integration Testing:
 - Purpose: To verify that different software modules and hardware components work together as intended.
 - Method: Conduct integration tests that involve the entire system—Raspberry Pi Pico W, MicroSD card interface, OLED display, and audio output through the DAC. This will include scripted tests to simulate user interactions and automated playback to verify the correct handling of audio data.
 
3. System Testing:
 - Purpose: To validate the complete and integrated software and hardware system against the requirements.
 - Method: Perform system-level testing to ensure the MP3 player operates under real-world conditions. This includes testing usability, audio quality, stability, and responsiveness.
 
4. User Acceptance Testing (UAT):
 - Purpose: To ensure the system meets the usability and functionality requirements from a user's perspective.
 - Method: Involve potential users in testing by providing them with the device to use in their daily routines. Collect feedback on user interface design, functionality, and overall user experience.

### TODO or Help Wanted

This pull request still needs:

- Feedback on the proposed system architecture and component integration.
- Suggestions for improvements or alternative approaches to the hardware setup and software stack.
- Review of the initial project documentation to ensure clarity and completeness.

### Build

- [ ] Ran `npm build`.

### Author

Signed-off-by: Popa Ruxandra-Georgiana <ruxandrapopa22@yahoo.com>

### CODE REVIEW

File: docs/project/2024/Ruxandra.popa/index.md
Comment (alexandruradovici) [inline docs/project/2024/Ruxandra.popa/index.md:20]:
```suggestion
### Schematic Diagram**
```

Please use the markdown format for subtitles.

File: docs/project/2024/Ruxandra.popa/index.md
Comment (alexandruradovici) [inline docs/project/2024/Ruxandra.popa/index.md:103]: These are C libraries, you have to use Rust libraries.

File: docs/project/2024/Ruxandra.popa/index.md
Comment (alexandruradovici) [inline docs/project/2024/Ruxandra.popa/index.md:104]: I think this lubrary is for the ESP8266 device which us not ARM.

**Review by alexandruradovici — CHANGES_REQUESTED** [review]

File: docs/project/2024/Ruxandra.popa/index.md
Comment (ruxi22) [reply docs/project/2024/Ruxandra.popa/index.md]: I have modified the libraries, thank you!

**Review by ruxi22 — COMMENTED** [review]
I have modified

File: docs/project/2024/Ruxandra.popa/index.md
Comment (alexandruradovici) [inline docs/project/2024/Ruxandra.popa/index.md:22]:
```suggestion
A detailed schematic diagram is provided to illustrate the connections and layout of the components. 

![schematic](Schematic.png)
```

**Review by alexandruradovici — CHANGES_REQUESTED** [review]

File: docs/project/2024/Ruxandra.popa/index.md
Comment (alexandruradovici) [inline docs/project/2024/Ruxandra.popa/index.md:24]: Please use the file from this repository.

**Review by alexandruradovici — CHANGES_REQUESTED** [review]

File: docs/project/2024/Ruxandra.popa/index.md
Comment (alexandruradovici) [inline docs/project/2024/Ruxandra.popa/index.md:24]:
```suggestion
![Schematic diagram](Schematic.png)
```

**Review by alexandruradovici — CHANGES_REQUESTED** [review]

**Review by alexandruradovici — DISMISSED** [review]

**Comment by ruxi22** [comment]
@alexandruradovici What should  I do with the Unresolved conversations and also with the Test deployment / Test deployment (pull_request). Is it all good now?

**Comment by alexandruradovici** [comment]
> @alexandruradovici What should  I do with the Unresolved conversations and also with the Test deployment / Test deployment (pull_request). Is it all good now?
> 
> 

Just click resolve.

**Comment by ruxi22** [comment]
@alexandruradovici 
I think I need another approval for the Review required(At least 2 approving reviews are required by reviewers with write access). and also Merging is blocked (Merging can be performed automatically with 2 approving reviews.).

**Comment by alexandruradovici** [comment]
That is ok, another person from the team must approve it.

**Comment by ruxi22** [comment]
> That is ok, another person from the team must approve it.

Ok, thank you !!

**Comment by ruxi22** [comment]
I just added the :::info :::: for the template,  I have deleted by mistake

**Review by alexandruradovici — APPROVED** [review]

**Review by DTeodor-Alexaandru — APPROVED** [review]

File: docs/project/2024/Ruxandra.popa/index.md
Comment (alexandruradovici) [inline docs/project/2024/Ruxandra.popa/index.md:2]: Please add here a one line description of the project.

**Review by alexandruradovici — CHANGES_REQUESTED** [review]

**Review by alexandruradovici — APPROVED** [review]

**Review by irina-b-dev — APPROVED** [review]

**Pull request merged**
